fhir-tools/src/main/java/com/ibm/fhir/tools/CodeGenerator.java

```diff
@@ -1475,6 +1477,154 @@ public int compare(JsonObject first, JsonObject second) {
+     * 
+     * @param elementDefinition
+     *            the element definition
+     * @param elementName
+     *            the element name, or $this
+     * @param valueSet
+     *            the value set
+     * @param strength
+     *            the binding strength
+     * @return the FHIRPath expression
+     */
+    private String generateVocabularyConstraintExpression(JsonObject elementDefinition, String elementName, String valueSet, String strength) {
+        StringBuilder sb = new StringBuilder();
+
+        String choiceText = "";
+        if (isChoiceElement(elementDefinition)) {
+            List<String> choiceTypeNames = getChoiceTypeNames(elementDefinition);
+            if (!choiceTypeNames.isEmpty()) {
+                String typeName = choiceTypeNames.get(0);
+                choiceText = ".as(" + typeName + ")";
+            }
+        }
+
+        // Generate constraint to element that has focus
+        if ("$this".equals(elementName)) {
+            sb.append(elementName).append(choiceText).append(".memberOf('").append(valueSet).append("', '").append(strength).append("')");
+        }
+        // Generate constraint to context element
+        else {
+            sb.append(elementName).append(choiceText);
+
+            if (isOptional(elementDefinition)) {
+                sb.append(".exists() implies (");
+            } else {
+                sb.append(".exists() and ");
+            }
+
+            sb.append(elementName).append(choiceText);
+
+            if (isRepeating(elementDefinition)) {
+                sb.append(".all(");
+            } else {
+                sb.append(".");
+            }
+
+            sb.append("memberOf('").append(valueSet).append("', '").append(strength).append("')");
+
+            if (isRepeating(elementDefinition)) {
+                sb.append(")");
+            }
+
+            if (isOptional(elementDefinition)) {
+                sb.append(")");
+            }
+        }
+
+        return sb.toString();
+    }
+
+    /**
+     * Gets the next generated constraint ID.
+     * 
+     * @param lastId
+     *            the last ID
+     * @param className
+     *            the model class name
+     * @return the next generated constraint ID
+     */
+    private String generateNextConstraintId(String lastId, String className) {
+        String nextSuffix = "0";
+        if (lastId != null) {
+            lastId = lastId.substring(lastId.indexOf("-") + 1);
+            if (Character.isLetter(lastId.charAt(lastId.length() - 1))) {
+                lastId = lastId.substring(0, lastId.length() - 1);
+            }
+            int nextKeyInt = Integer.parseInt(lastId) + 1;
+            nextSuffix = String.valueOf(nextKeyInt);
+        }
+        return camelCase(className) + "-" + nextSuffix;
+    }
+
+    /**
+     * Determines if any of the structure definition contains any elements with an extensible or preferred binding.
```

Comment:
remove 'any of' before 'the structure definition'

Reply:
Fixed.